bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java

```diff
@@ -0,0 +1,656 @@
+                        if (result instanceof RadioThermostatJsonRuntime) {
+                            rthermData.setRuntime((RadioThermostatJsonRuntime) result);
+                        }
+                    }
+                } catch (Exception e) {
+                    logger.error("Exception occurred during execution: {}", e.getMessage(), e);
+                }
+            };
+
+            int delay = (config.logRefresh != null) ? config.logRefresh.intValue() : DEFAULT_LOG_REFRESH_PERIOD;
+            logRefreshJob = scheduler.scheduleWithFixedDelay(runnable, 1, delay, TimeUnit.MINUTES);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the RadioThermostat handler.");
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+        if (logRefreshJob != null && !logRefreshJob.isCancelled()) {
+            logRefreshJob.cancel(true);
+            logRefreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId(), rthermData);
+        } else {
+            Integer cmdInt = -1;
+            // remove all non-numeric characters except negative '-'
+            String cmdStr = command.toString().replaceAll("[^\\d-]", "");
+            if (!StringUtils.isEmpty(cmdStr) && StringUtils.isNumeric(cmdStr)) {
+                cmdInt = Integer.parseInt(cmdStr);
+            }
+            
+            switch (channelUID.getId()) {
+                case JSON_CMD:
+                    sendCommand(null, null, command.toString());
+                    break;
+                case MODE:
+                    //only do if commanded mode is different than current mode
+                    if (!cmdInt.equals(rthermData.getThermostatData().getMode())) {
+                        sendCommand("tmode", cmdStr);
+                        
+                        // set the new operating mode, reset everything else,
+                        // because refreshing the tstat data below is really slow.
+                        rthermData.getThermostatData().setMode(cmdInt);
+                        rthermData.getThermostatData().setHeatTarget(null);
+                        rthermData.getThermostatData().setCoolTarget(null);
+                        rthermData.getThermostatData().setHold(0);
+                        rthermData.getThermostatData().setProgramMode(-1);
+                        updateAllChannels();
+                        
+                        //now just go ahead and refresh tstat data to update with the new active setpoint
+                        // this takes a while for the JSON request to complete.
+                        Object result = getRadioThermostatData(DEFAULT_RESOURCE);
```

Comment:
Consider doing this asynchronously (see Jetty link above), to not block a openHAB's scheduler Thread.